gui/controls/HeaderButton.qml

```diff
@@ -0,0 +1,21 @@
+import QtQuick 2.11
+import QtQuick.Controls 2.4
+import ".."
+
+/*!
+    \qmltype HeaderButton
+    \brief A border-base button used in main header of the screen.
```

Comment:
Are we going to use this for more stuff than mode selection? If it's just mode selection, then maybe rename to ModeSelectionButton?

Reply:
This is outside ModeSelection, in the "header" part. (Where you click mode selection, settings, and so on). Again, because the design is changing I am not sure if that will stay for long. But Keeping a single element makes easy to change later.